### ISSUE TITLE
Bug: [no-unnecessary-type-assertion] should report `readonly` class properties with a literal initializer

### ISSUE THREAD

**Issue by ronami** [typescript-eslint/typescript-eslint#10617]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.7.2&fileType=.ts&code=C4TwDgpgBAylC8UCMAGA3AKAwem1AThGAPb7AQAmUAxqYdcADYga0B2AzsFAIYJQByHgN4caxTsEzsuUAEb8AzKKiLpE2dX6oVMTK0Y8OYgCpQA3hihRcUDgAtiAV0ZU50QiTKUA-FYIQPBQSzLz8QiJG4pL%2BhEEhIPJKKoqxgcFsoVqIOlEwGAC%2BQA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1tiacTJTIAhtEK0ipWkOTJE0fJQ5N0UOdA7RI4MAF8QOoA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
type S = 10;

// reported correctly
const a = 'a' as const;
const b = 3 as 3;
const c = 10 as S;

class T {
  // should be reported?
  readonly a = 'a' as const
  readonly b = 3 as 3
  readonly c = 10 as S
}
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/no-unnecessary-type-assertion": "error",
  },
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": true
  }
}
```

### Expected Result

I expected these to be reported, similarly to the `const` declarations.

### Actual Result

The various `readonly` declarations haven't been reported.
